components/knowledge/Notion.tsx

```diff
@@ -0,0 +1,229 @@
+  };
+
+  const handleSelectedFileChange = (selected: any) => {
+    if (selected === 'all') {
+      setSelectedFile(Array.from(notionFiles.keys()));
+    } else {
+      setSelectedFile(Array.from(selected as Set<string>));
+    }
+  };
+
+  return (
+    <Modal
+      size="4xl"
+      isOpen={isOpen}
+      onClose={onClose}
+      hideCloseButton={true}
+      scrollBehavior="inside"
+      className="min-h-[40vh] max-h-[80vh]"
+    >
+      <ModalContent>
+        <ModalHeader>
+          <div className="w-full flex items-center justify-between">
+            <div className="flex flex-row items-center">
+              <Image className="h-6 w-6" src="notion.svg" alt="Notion Icon" />
+              <p className="ml-2">Notion</p>
+            </div>
+
+            <div className="flex items-center justify-end p-2">
+              <Button
+                isLoading={isSyncing}
+                size="sm"
+                color="primary"
+                startContent={!isSyncing && <IoMdRefresh />}
+                onClick={syncNotion}
+              >
+                {!isSyncing && 'Sync'}
+              </Button>
+              {syncError && (
+                <p className="text-sm text-red-500 ml-2">{syncError}</p>
+              )}
+            </div>
+          </div>
+        </ModalHeader>
+        <ModalBody>
+          <Input
+            className="w-[20%] flex justify-end"
+            placeholder="Search"
+            size="sm"
+            value={searchQuery}
+            onChange={(e) => {
+              setSearchQuery(e.target.value);
+            }}
+            startContent={<CiSearch />}
+          />
+          {notionConfigured && notionFiles.size > 0 ? (
+            <div className="flex flex-col gap-1">
+              <Table
+                selectionMode="multiple"
+                selectionBehavior="toggle"
+                aria-label="notion-files"
+                isCompact={true}
+                selectedKeys={selectedFile}
+                onSelectionChange={(selected) => {
+                  handleSelectedFileChange(selected);
+                }}
+              >
+                <TableHeader>
+                  <TableColumn>NAME</TableColumn>
+                  <TableColumn>Link</TableColumn>
+                </TableHeader>
+                <TableBody>
+                  {Array.from(notionFiles.entries())
+                    .filter(([_, file]) => {
+                      if (!searchQuery) return true;
+                      return file.fileName
+                        .toLowerCase()
+                        .includes(searchQuery.toLowerCase());
+                    })
+                    .map(([key, value]) => (
+                      <TableRow key={key}>
+                        <TableCell>
+                          <div className="flex flex-col">
+                            <p>{value.fileName}</p>
+                          </div>
+                        </TableCell>
+                        <TableCell>
```

Comment:
super nit:

This makes it so that the link doesn't take up a ton of space.
```suggestion
                        <TableCell className="w-[80px]">
```

<details>
<summary>It looks like this</summary>

<img width="886" alt="Screenshot 2024-09-05 at 4 59 31 PM" src="https://github.com/user-attachments/assets/965f588e-bc75-445a-b548-a506f61a85b3">


</details>